Bigtable/src/BigtableClient.php

```diff
@@ -0,0 +1,196 @@
+                'callable' => function ($instance) {
+                    $name = InstanceAdminClient::parseName($instance['name'])['instance'];
+                    return $this->instance($name, $instance);
+                }
+            ], [
+                'typeUrl' => 'type.googleapis.com/google.bigtable.admin.instance.v2.CreateInstanceMetadata',
+                'callable' => function ($instance) {
+                    $name = InstanceAdminClient::parseName($instance['name'])['instance'];
+                    return $this->instance($name, $instance);
+                }
+            ]
+        ]);
+    }
+
+    /**
+     * Lazily instantiate an instance.
+     *
+     * Example:
+     * ```
+     * $instance = $bigtable->instance('my-instance');
+     * ```
+     *
+     * @param string $instanceId The instance ID
+     * @param array $instance [optional] {
+     *     Configuration options
+     *
+     *     @type string $displayName **Defaults to** the value of $instanceId.
+     *     @type int $instanceType Possible values are represented by the following constants:
+     *           `Google\Cloud\Bigtable\Instance::INSTANCE_TYPE_PRODUCTION` and
+     *           `Google\Cloud\Bigtable\Instance::INSTANCE_TYPE_DEVELOPMENT`.
+     *           **Defaults to** using `Google\Cloud\Bigtable\Instance::INSTANCE_TYPE_UNSPECIFIED`.
+     *     @type array $labels as key/value pair ['foo' => 'bar']. For more information, see
+     *           [Using labels to organize Google Cloud Platform resources](https://cloudplatform.googleblog.com/2015/10/using-labels-to-organize-Google-Cloud-Platform-resources.html).
+     * }
+     * @return Instance
+     */
+    public function instance($instanceId, array $instance = [])
+    {
+        return new Instance(
+            $this->connection,
+            $this->lroConnection,
+            $this->lroCallables,
+            $this->projectId,
+            $instanceId,
+            $instance
+        );
+    }
+
+    /**
+     * Constructs an array that contains cluster metadata.
+     *
+     * Example:
+     * ```
+     * $cluster = $bigtable->clusterMetadata();
+     * ```
+     * @param string $clusterId The cluster ID
+     * @param string $locationId The location ID
+     * @param int $storageType The storage media type for persisting Bigtable data. Possible values include
+     *        `Google\Cloud\Bigtable\Instance::STORAGE_TYPE_SSD`,
+     *        `Google\Cloud\Bigtable\Instance::STORAGE_TYPE_HDD` and
+     *        `Google\Cloud\Bigtable\Instance::STORAGE_TYPE_UNSPECIFIED`.
+     *        **Defaults to** `Google\Cloud\Bigtable\Instance::STORAGE_TYPE_UNSPECIFIED`.
+     * @param int $serveNodes The number of nodes allocated to this cluster.
+     *        More nodes enable higher throughput and more consistent performance.
+     * }
+     * @return array
+     */
+    public function clusterMetadata(
```

Comment:
I'm trying to understand why we need the params to be nullable.
@jdpedrie  @dwsupplee 
Is this something you guys suggested? If so, would you mind explaining a bit?
The original version didn't allow null values, which I think it made more sense, as this is aimed to be a helper function for people. So, at least we should guide them in the correct way--if cluster ID is required for all situations, shouldn't we disallow it to be null at first?

Reply:
I think you're right. `clusterId` and `locationId` are required by `Instance::create()`. Therefore, those arguments should be required on this method.

Reply:
I agree, it makes more sense for them to be required.